### ISSUE TITLE
Installing AllenNLP in fresh conda env has errors

### ISSUE THREAD

**Issue by dericp** [allenai/allennlp#2882]
**Describe the bug**
When installing AllenNLP in a fresh Anaconda env, there are several errors.

**To Reproduce**
```
conda create --name my-env
conda install pip
pip install allennlp --no-cache-dir
```

At this point, I see some errors:
```
ERROR: botocore 1.12.155 has requirement urllib3<1.25,>=1.20; python_version >= "3.4", but you'll have urllib3 1.25.3 which is incompatible.
ERROR: aws-sam-translator 1.11.0 has requirement jsonschema~=2.6, but you'll have jsonschema 3.0.1 which is incompatible.
ERROR: cfn-lint 0.21.0 has requirement jsonschema~=2.6, but you'll have jsonschema 3.0.1 which is incompatible.
ERROR: cfn-lint 0.21.0 has requirement requests<=2.21.0,>=2.15.0, but you'll have requests 2.22.0 which is incompatible.
```
These can be fixed manually:

```
pip install urllib3==1.24.3
pip install jsonschema==2.6
pip install requests==2.21.0
```

Then, running `allennlp test-install` fails two tests.
Flaky test report:
```
===Flaky Test Report===

test_evaluate_from_args passed 1 out of the required 1 times. Success!
test_atis_model_can_train_save_and_load passed 1 out of the required 1 times. Success!
test_batch_predictions_are_consistent passed 1 out of the required 1 times. Success!
test_batch_predictions_are_consistent passed 1 out of the required 1 times. Success!
test_model_can_train_save_and_load passed 1 out of the required 1 times. Success!
test_batch_predictions_are_consistent passed 1 out of the required 1 times. Success!
test_batch_predictions_are_consistent passed 1 out of the required 1 times. Success!
test_batch_predictions_are_consistent passed 1 out of the required 1 times. Success!
test_model_can_train_save_and_load passed 1 out of the required 1 times. Success!
test_model_can_train_save_and_load failed (1 runs remaining out of 2).
        <class 'Exception'>
        Incorrect gradients found. See stdout for more info.
        [<TracebackEntry /home/dericp/anaconda3/envs/syntactic-entailment/lib/python3.7/site-packages/allennlp/tests/models/reading_comprehension/naqanet_test.py:16>, <TracebackEntry /home/dericp/anaconda3/envs/syntactic-entailment/lib/python3.7/site-packages/allennlp/com
mon/testing/model_test_case.py:106>, <TracebackEntry /home/dericp/anaconda3/envs/syntactic-entailment/lib/python3.7/site-packages/allennlp/common/testing/model_test_case.py:198>]
test_model_can_train_save_and_load failed; it passed 0 out of the required 1 times.
        <class 'Exception'>
        Incorrect gradients found. See stdout for more info.
        [<TracebackEntry /home/dericp/anaconda3/envs/syntactic-entailment/lib/python3.7/site-packages/allennlp/tests/models/reading_comprehension/naqanet_test.py:16>, <TracebackEntry /home/dericp/anaconda3/envs/syntactic-entailment/lib/python3.7/site-packages/allennlp/com
mon/testing/model_test_case.py:106>, <TracebackEntry /home/dericp/anaconda3/envs/syntactic-entailment/lib/python3.7/site-packages/allennlp/common/testing/model_test_case.py:198>]
test_model_can_train_save_and_load passed 1 out of the required 1 times. Success!
test_atis_parser_batch_predicted_sql_present passed 1 out of the required 1 times. Success!
test_atis_parser_predicted_sql_present passed 1 out of the required 1 times. Success!
test_atis_parser_uses_named_inputs passed 1 out of the required 1 times. Success!

===End Flaky Test Report===
```
The full output of `allennlp test-install` is attached: [test-install-out.txt](https://github.com/allenai/allennlp/files/3214533/test-install-out.txt)

**Expected behavior**
No errors when installing.

**System (please complete the following information):**
 - OS: Linux, Ubuntu
 - Python version: 3.7.3
 - AllenNLP version: 0.8.3
 - PyTorch version: 1.1.0

**Additional context**
After EMNLP, I wanted to see how my models would run with the latest PyTorch and AllenNLP versions when I discovered that trying to update these packages broke many things. Then, I tried creating a new environment to run my models and could not.

**Comment by joelgrus** [comment]
I can reproduce the pip install errors (it looks like they're caused by a newer version of `moto`, we may have to add an upper bound on that, although it's not clear to me that they affect anything adversely), but I cannot reproduce the test failures.

if you try running those two tests individually do they still fail?

**Comment by dericp** [comment]
Running `allennlp test-install -k test_output_only_elmo_bcn_can_train_save_and_load` gives this output 
[output1.txt](https://github.com/allenai/allennlp/files/3218553/output1.txt).
Running `allennlp test-install -k naqanet_test` gives this output 
[output2.txt](https://github.com/allenai/allennlp/files/3218555/output2.txt).

Looks like they both fail when running individually.

**Comment by nelson-liu** [comment]
to chime in, since i implemented the BCN:

I've seen that error on the BCN test too, intermittently. I think I can reproduce it on OS X, but I've never seen it on a linux machine (with or without GPU).

**Comment by joelgrus** [comment]
We think these zero gradients might be caused by either dropout in the test models or by the activation functions, we'll make changes to those tests, but it's probably not anything you need to worry about for your actual code.

**Comment by dericp** [comment]
Sounds good---thanks for the quick response.

**Comment by gvskalyan** [comment]
pip install --editable .    in fresh conda python3.6 / python3.7 env gives the following errors and allennlp is not installed as it gives pkg_resources.ContextualVersionConflict: (requests 2.22.0 (, Requirement.parse('requests<=2.21.0,>=2.15.0'), {'cfn-lint'})))

**Comment by kl2806** [comment]
@gvskalyan that should be fixed by PR #2902 .
